doc/running-simple-app.md

```diff
@@ -0,0 +1,209 @@
+
+To support these features, Mystikos runs the following modules within the
+trusted execution environment (TEE).
+
+- EXT2 file system driver (ext2)
+- Block integrity checking device (dm-verity)
+- Block encryption device (dm-crypt)
+- Block cache device
+
+The EXT2 image itself resides on the untrusted file system. The TEE transitions
+into the untrusted execution environment to fetch blocks. The blocks are then
+decrypted and integrity-checked within the TEE.
+
+EXT2 images generated by Mystikos has the following layout, divided into three
+sections.
+
+| EXT2 file system | hash device | file-system signature (fssig) |
+| ---------------- |:-----------:| -----------------------------:|
+
+The first section is a Linux-compatible EXT2 file system. The next section is
+a Linux-compatible dm-verity hash device (Merkle tree). The final section is
+a 4096-byte proprietary structure that defines the **root hash** of the
+Merkle tree and an optional **digital signature** of the root hash.
+
+### Host file system (insecure)
+
+Mystikos supports mounting directories from the host file system. This option is not secured
+by the enclave, but is useful for applications that need to manipulate untrusted data on the 
+host.
+
+> Mystikos applications may use a mixture of file system types and mount types.
+
+Creating a Simple Application Image
+----------------------
+
+This section explains how to create single-file images for use with
+Mystikos.
+
+### CPIO
+
+The following command creates a CPIO image (``cpioimage``) from the
+contents of a directory (``appdir``).
+
+```
+$ myst mkcpio appdir cpioimage
+```
+### EXT2
+
+The following command creates an EXT2 image (``ext2image``) from the
+contents of a directory (``appdir``).
+
+```
+$ myst mkext2 appdir ext2image
+```
+
+> To create a signed EXT2 image to use with a signed application package, see [sign-package.md](./sign-package.md)
+
+Loading a root file system
+--------------------------------
+
+This section explains how to use an application image as a Mystikos root file system
+(or rootfs).
+
+### Using an application image during development
+During development, an application image is passed to the ``myst exec`` command as
+follows.
+
+```
+$ myst exec-sgx <rootfs> /bin/hello
+Hello World!
+```
+
+> `<rootfs>` can be a CPIO image, an EXT2 image, or a hostfs directory name.
+
+Advanced Features
+------------
+## Creating an encrypted EXT2 image 
+
+The following command creates an encrypted EXT2 image.
+
+```
+$ myst mkext2 --encrypt=keyfile ext2image
+```
+
+Use the following command generate a test keyfile.
+
+```
+$ head -c 64 /dev/urandom > keyfile
+```
+
+The EXT2 image created above can be decrypted by a Mystikos image that possesses
+the key. Key distribution is left to the application. An encrypted image cannot
+be used as the root file system.
+
+##  Mounting EXT2 images from applications (explicit mount or "app mounted")
+
```

Comment:
This section doesn't belong to a `running simple app` document. Explicit mount is not recommended in general as it's unsafe and it is not suited for beginners. Consider shortening the section considerably or remove it altogether.

Reply:
I decided to remove this section. Roothash/pubkey is also not checked in explicit mount.